spec/amazon_ses_mailer/base_spec.rb

```diff
@@ -1,79 +1,57 @@
       end
     end
-  
-    describe '.deliveries' do
-      context 'when there are no deliveries' do
-        it 'should return empty array' do
-          expect(AmazonSesMailer::Base.deliveries).to eq([])
-        end
-      end
-  
-      context 'when there are deliveries' do
-        let(:delivery_method) { :test }
-        before do
-          allow(AmazonSesMailer::Base).to receive(:delivery_method)
-            .and_return(delivery_method)
-        end
-        it {
-          expect { AmazonSesMailer::Base.mail(options).deliver }
-            .to change { AmazonSesMailer::Base.deliveries.count }.by(1)
-        }
+
+    context 'when an email is delivered' do
+      options = { from: 'user@example.com', to: 'user@example.com' }
+      AmazonSesMailer::Base.delivery_method = :test
+      it 'should not have zero email deliveries' do
+        AmazonSesMailer::Base.mail(options).deliver
+        expect(AmazonSesMailer::Base.deliveries).not_to eq([])
       end
     end
-    describe '.transform_array' do
-      context 'when given object is array' do
-        it 'should convert false/nil into empty string and all others into string' do
-          arr = [{ a: 1 }, false]
-          expect(AmazonSesMailer::Base.transform_array(arr)).to eq([{ a: '1' }, ''])
-        end
-      end
-  
-      context 'when given object is not array' do
-        it 'should not convert false/nil into empty string and all others into string' do
-          arr = { a: 1, b: false }
-          expect(AmazonSesMailer::Base.transform_array(arr)).not_to eq([{ a: '1' }, ''])
-        end
+  end
+
+  describe '.register_inceptors' do
+    context 'when there is interceptor.delivering_email to register interceptor' do
+      it 'should register interceptors' do
+        expect(AmazonSesMailer::Base.register_interceptor(MyInterceptor)).to eq([MyInterceptor])
       end
     end
-    describe '.register_interceptor' do
-      context 'when there are interceptors to block certain email deliveries' do
-        class MyInterceptor
-          def self.delivering_email(message)
-            message[:destination][:to_addresses].any? do |email|
-              email.include?('@example.com')
-            end
-          end
-        end
-  
-        it 'should register those interceptors' do
-          expect(AmazonSesMailer::Base.register_interceptor(MyInterceptor)).to eq([MyInterceptor])
-        end
-      end
-  
-      context 'when there are is no delivering_mail to respond' do
+
+    context 'when there is no interceptor.delivering_email method to register incerceptor' do
+      before do
         class FalseInterceptor
         end
-        it 'should raise error' do
-          expect { AmazonSesMailer::Base.register_interceptor(FalseInterceptor) }.to raise_error
-        end
       end
+      it 'should not respond to delivering_email' do
+        expect { AmazonSesMailer::Base.register_interceptor(FalseInterceptor) }.to raise_error(RuntimeError)
```

Comment:
This is great, but it is better to specify the error message here to ensure the correct error is raised.